### DISCUSSION TITLE
using switchmode ?

### DISCUSSION THREAD

**Discussion by defaliz** [arendst/Tasmota#14616] [Q&A] [answered]
hi..
i don't know if i understand perfectly the switchmode notion...
switchmode is only for a physical switch connected to a gpio ? am i right ?

but with relay (a button on web gui) configured in tasmota on a gpio, how to have HOLD button action ?
thank for replies...
Alain

**Comment by sfromis** [comment, marked as answer]
SwitchMode is for an input pin configured as Switch, typically connected to a pushbutton, toggle switch or PIR. For basic actions (including HOLD), configuring as Button instead of Switch also makes sense, but then SwitchMode is not used for config.

Relay is configured on an output pin, which indeed also results in a button on the web UI, but has no HOLD action or similar. You cannot get the features of input pin (Switch/Button) on the web UI button. Actions like HOLD are tied to having an actual input pin.

**Comment by defaliz** [comment]
Very clear, Thank sfromis..
In the future it will be great to have hold action on a web ui button..because with a smartphone it easy to push quickly  on a button accidentaly..
